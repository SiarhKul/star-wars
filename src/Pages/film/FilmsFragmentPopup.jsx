import React from "react";
import { useSelector } from "react-redux";
import { useLocation } from "react-router";
import PopupListItem from "../../components/popup/PopupListItem";
import { clickedCard } from "../../utils";

export const FilmsFragmentPopup = () => {
	const films = useSelector(state => state.dataFromServer.films);
	const query = new URLSearchParams(useLocation().search);
	const paramField = query.get("selected");
	const card = clickedCard(films, paramField, "title");
	const {
		title,
		episode_id,
		director,
		producer,
		release_date,
		characters,
		planets,
		vehicles,
		starships,
		species,
	} = card;

	return (
		<div className="popup-fragment">
			<div className="name-wrapper">
				<span className="popup__name">{title}</span>
			</div>
			<div className="description-wrapper">
				<div className="description-person">
					<div className="description-appearance">
						<p className="description__heading">Appearance</p>
						<ul className="description-appearance__item">
							<li>Episode:{episode_id}</li>
							<li>Director:{director}</li>
							<li>Producer:{producer}</li>
							<li>Release date:{release_date}</li>
						</ul>
					</div>
				</div>

				<div className="description-resources">
					<div className="resources">
						<div className="resources-heading">Characters:</div>
						<ul className="resources-list">
							<PopupListItem urls={characters} name="name" path="people" />
						</ul>
					</div>

					<div className="resources">
						<div className="resources-heading">Planets:</div>
						<ul className="resources-list">
							<PopupListItem urls={planets} name="name" path="planets" />
						</ul>
					</div>

					<div className="resources">
						<div className="resources-heading">Vehicles:</div>
						<ul className="resources-list">
							<PopupListItem urls={vehicles} name="name" path="vehicles" />
						</ul>
					</div>

					<div className="resources">
						<span className="resources-heading">Starships:</span>
						<ul className="resources-list">
							<PopupListItem urls={starships} name="name" path="starships" />
						</ul>
					</div>
					<div className="resources">
						<span className="resources-heading">Species:</span>
						<ul className="resources-list">
							<PopupListItem urls={species} name="name" path="species" />
						</ul>
					</div>
				</div>
			</div>
		</div>
	);
};
