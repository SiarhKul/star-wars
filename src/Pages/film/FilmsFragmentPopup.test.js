import React from 'react';
import { shallow } from 'enzyme';

import { FilmsFragmentPopup } from './FilmsFragmentPopup';

jest.mock('history', () => ({
  createBrowserHistory: jest.fn(() => ({
    location: {
      state: {
        title: 'A New Hope',
        episode_id: 3,
        opening_crawl:
						"It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire's\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire's\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....",
        director: 'George Lucas',
        producer: 'Gary Kurtz, Rick McCallum',
        release_date: '1977-05-25',
        characters: ['https://swapi.dev/api/people/1/'],
        planets: ['https://swapi.dev/api/planets/1/'],
        starships: ['https://swapi.dev/api/starships/2/'],
        vehicles: ['https://swapi.dev/api/vehicles/4/'],
        species: ['https://swapi.dev/api/species/1/'],
      },
    },
  })),
}));

describe('Test <FilmsFragmentPopup/>', () => {
  const filmsFragmentPopup = () => shallow(<FilmsFragmentPopup />);
  const component = filmsFragmentPopup();

  it('should render <FilmsFragmentPopup/>  component ', () => {
    expect(component).toMatchSnapshot();
  });

  it('should <PeopleFragmentPopup/> have heading', () => {
    const requiredFields = [
      'Films',
      'Episode:',
      'Director:',
      'Producer:',
      'Release date:',
      'Characters:',
      'Planets:',
      'Vehicles:',
      'Starships:',
      'Species:',
    ];
    expect(component.containsAllMatchingElements(requiredFields)).toBeTruthy();
  });
});
