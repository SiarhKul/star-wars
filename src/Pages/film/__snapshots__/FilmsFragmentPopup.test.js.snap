// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FilmsFragmentPopup should render FilmsFragmentPopup component  1`] = `
<div
  className="popup-fragment"
>
  <div
    className="name-wrapper"
  >
    <span
      className="popup__name"
    >
      A New Hope
    </span>
  </div>
  <div
    className="description-wrapper"
  >
    <div
      className="description-person"
    >
      <div
        className="description-appearance"
      >
        <p
          className="description__heading"
        >
          Appearance
        </p>
        <ul
          className="description-appearance__item"
        >
          <li>
            Episode:
            3
          </li>
          <li>
            Director:
            George Lucas
          </li>
          <li>
            Producer:
            Gary Kurtz, Rick McCallum
          </li>
          <li>
            Release date:
            1977-05-25
          </li>
        </ul>
      </div>
    </div>
    <div
      className="description-resources"
    >
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Characters:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="people"
            urls={
              Array [
                "https://swapi.dev/api/people/1/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Planets:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="planets"
            urls={
              Array [
                "https://swapi.dev/api/planets/1/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <div
          className="resources-heading"
        >
          Vehicles:
        </div>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="vehicles"
            urls={
              Array [
                "https://swapi.dev/api/vehicles/4/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <span
          className="resources-heading"
        >
          Starships:
        </span>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="starships"
            urls={
              Array [
                "https://swapi.dev/api/starships/2/",
              ]
            }
          />
        </ul>
      </div>
      <div
        className="resources"
      >
        <span
          className="resources-heading"
        >
          Species:
        </span>
        <ul
          className="resources-list"
        >
          <PopupListItem
            name="name"
            path="species"
            urls={
              Array [
                "https://swapi.dev/api/species/1/",
              ]
            }
          />
        </ul>
      </div>
    </div>
  </div>
</div>
`;
